### ISSUE TITLE
LGPL dependencies packaged with druid main tarball

### ISSUE THREAD

**Issue by b-slim** [druid-io/druid#3866]
We have some core dependency that brings unwanted LGPL code.
```
INFO] |  \- io.airlift:airline:jar:0.7:compile
[INFO] |     \- com.google.code.findbugs:annotations:jar:2.0.3:compile
```
https://mvnrepository.com/artifact/com.google.code.findbugs/annotations/2.0.3

**Comment by b-slim** [comment]
fix on https://github.com/druid-io/druid/pull/3865

**Comment by nishantmonu51** [comment]
Since we have accidentally shipped this jar with 0.9.2 and prev releases and polluted them with GPL code. I think we should also do a 0.9.2.1 release with this fix included.
@cheddar @gianm @fjy @drcrallen @himanshug Any thoughts ?

**Comment by drcrallen** [comment]
https://mvnrepository.com/artifact/com.google.code.findbugs/annotations/2.0.3 says `LGPL` not `GPL`

Doing a release is cheap, and if it alleviates concerns I say we go for it if someone is able to write up the short release notes.

**Comment by fjy** [comment]
@b-slim LGPL is fine

**Comment by fjy** [comment]
@b-slim I'm pretty sure the last time we did an audit pre 0.9.2 release there were a few LGPL projects or GPL with cp exception so I don't believe there's any pollution

**Comment by b-slim** [comment]
> LGPL is fine


@fjy thought that the diff between GPL vs. LGPL determines whether you do/do not have to release the source for your application to anyone having a copy of the binary, but it doesn't say anything about  does or does not taint the apache license any ref will be useful ?

**Comment by cheddar** [comment]
LGPL only requires release of source if you actually change the content of the LGPL code.  Just including a pre-compile copy of the code doesn't taint.

**Comment by gianm** [comment]
My  understanding is that opinions differ on LGPL but the ASF's position is clear that it is forbidden in ASF projects (see https://www.apache.org/legal/resolved). Druid is _not_ an ASF project but for consistency we might as well get rid of LGPL dependencies.

I don't think it's necessary to do a 0.9.2.1 release but it makes sense to get rid of these dependencies for our next release.

**Comment by cheddar** [comment]
I'll buy that argument @gianm

**Comment by nishantmonu51** [comment]
Agree with @gianm, No release required still better to remove it.

**Comment by gianm** [comment]
fixed in #3865.
